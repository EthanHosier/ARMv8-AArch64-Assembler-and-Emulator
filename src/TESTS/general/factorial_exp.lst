
test/expected/test/test_cases/general/factorial_exp.bin:     file format binary


Disassembly of section .data:

0000000000000000 <.data>:
   0:	d2800020 	mov	x0, #0x1                   	// #1
   4:	d28000a1 	mov	x1, #0x5                   	// #5
   8:	9b007c22 	mul	x2, x1, x0
   c:	91000040 	add	x0, x2, #0x0
  10:	d1000421 	sub	x1, x1, #0x1
  14:	f100003f 	cmp	x1, #0x0
  18:	54ffff81 	b.ne	0x8  // b.any
  1c:	d2802003 	mov	x3, #0x100                 	// #256
  20:	f9000062 	str	x2, [x3]
  24:	8a000000 	and	x0, x0, x0
