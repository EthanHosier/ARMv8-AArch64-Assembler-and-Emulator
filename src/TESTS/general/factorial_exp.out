Registers:
X00    = 0000000000000078
X01    = 0000000000000000
X02    = 0000000000000078
X03    = 0000000000000100
X04    = 0000000000000000
X05    = 0000000000000000
X06    = 0000000000000000
X07    = 0000000000000000
X08    = 0000000000000000
X09    = 0000000000000000
X10    = 0000000000000000
X11    = 0000000000000000
X12    = 0000000000000000
X13    = 0000000000000000
X14    = 0000000000000000
X15    = 0000000000000000
X16    = 0000000000000000
X17    = 0000000000000000
X18    = 0000000000000000
X19    = 0000000000000000
X20    = 0000000000000000
X21    = 0000000000000000
X22    = 0000000000000000
X23    = 0000000000000000
X24    = 0000000000000000
X25    = 0000000000000000
X26    = 0000000000000000
X27    = 0000000000000000
X28    = 0000000000000000
X29    = 0000000000000000
X30    = 0000000000000000
PC     = 0000000000000024
PSTATE : -ZC-
Non-Zero Memory:
0x00000000 : d2800020
0x00000004 : d28000a1
0x00000008 : 9b007c22
0x0000000c : 91000040
0x00000010 : d1000421
0x00000014 : f100003f
0x00000018 : 54ffff81
0x0000001c : d2802003
0x00000020 : f9000062
0x00000024 : 8a000000
0x00000100 : 00000078
